README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# SWATH2stats

This package is intended to transform extracted SWATH/DIA data from the
OpenSWATH or other (e.g. Spectronaut) software into a format
directly-usable by statistics packages (e.g. mapDIA, PECA, MSstats)
while performing filtering, annotation and FDR assessment if necessary.

## Analyzing SWATH/DIA data

How to extract SWATH/DIA data before using SWATH2stats with OpenSWATH
can be found here: <http://openswath.org>

## Usage

SWATH2stats is a Bioconductor package. Go to
<https://www.bioconductor.org/packages/release/bioc/html/SWATH2stats.html>
to see all information related to installation. Importantly there exists
both a release and development version.

## Contribution

Please feel free to comment and post issues or pull requests on github.

## Publication

For the publication describing this package, see:
<https://doi.org/10.1371/journal.pone.0153160>

## References

  - Blattmann P, Heusel M, Aebersold R. SWATH2stats: An R/Bioconductor
    Package to Process and Convert Quantitative SWATH-MS Proteomics Data
    for Downstream Analysis Tools. PLoS ONE 11(4): e0153160 (2016). doi:
    10.1371/journal.pone.0153160.

  - Rost HL, Rosenberger G, Navarro P, Gillet L, Miladinovic SM,
    Schubert OT, Wolski W, Collins BC, Malmstrom J, Malmstrom L,
    Aebersold R. OpenSWATH enables automated, targeted analysis of
    data-independent acquisition MS data. Nature Biotechnology. 2014
    Mar;32(3):219-23. doi: 10.1038/nbt.2841.

  - Choi M, Chang CY, Clough T, Broudy D, Killeen T, MacLean B, Vitek O.
    MSstats: an R package for statistical analysis of quantitative mass
    spectrometry-based proteomic experiments.Bioinformatics. 2014 Sep
    1;30(17):2524-6. doi: 10.1093/bioinformatics/btu305.

  - Rosenberger G, Ludwig C, Rost HL, Aebersold R, Malmstrom L. aLFQ: an
    R-package for estimating absolute protein quantities from label-free
    LC-MS/MS proteomics data. Bioinformatics. 2014 Sep 1;30(17):2511-3.
    doi: 10.1093/bioinformatics/btu200.

  - Suomi T., Corthals G., Nevalainen O.S., and Elo L.L. (2015). Using
    Peptide-Level Proteomics Data for Detecting Differentially Expressed
    Proteins. J Proteome Res. Nov 6;14(11):4564-70. doi:
    10.1021/acs.jproteome.5b00363.

  - Suomi, T. and Elo L.L. (2017). Enhanced differential expression
    statistics for data-independent acquisition proteomics" Scientific
    Reports 7, Article number: 5869.doi:10.1038/s41598-017-05949-y


NEWS
CHANGES IN VERSION 1.20.1
-------------------------
BUG FIXES

 o add rmarkdown to DESCRIPTION



CHANGES IN VERSION 1.20.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.12 release


CHANGES IN VERSION 1.19.3
-------------------------
UPDATE

 o Add link to website
 


CHANGES IN VERSION 1.19.2
-------------------------
BUG FIXES

 o remove grid from Description
 


CHANGES IN VERSION 1.19.1
-------------------------
NEW FEATURES

 o validate_columns
 
update
 
 o update documentation to roxygen2 (many thanks to Ashton Trey Belew (abelew) for starting this and doing most of the work) (IS#1, PR#2, PR#3)
 
 o Some parameters were renamed to omit having a dot. E.g. rm.decoy is now called rm_decoy. If you want the previous names use SWATH2stats up to version 1.19.0
 
 



 

CHANGES IN VERSION 1.18.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.11 release


CHANGES IN VERSION 1.17.5
-------------------------
update

 o .Rmd vignettes: remove tinytex option


CHANGES IN VERSION 1.17.4
-------------------------
update

 o convert_protein_ids: Also count empty strings as NA and count unique non-mapped identifiers.


CHANGES IN VERSION 1.17.3
-------------------------
update

 o plot_correlation_between_samples.R, plot_variation.R, plot_variation_vs_total.R: improve plots and do theme_bw



CHANGES IN VERSION 1.17.2
-------------------------
NEW FEATURES

 o filter_on_min_peptides.R: add ability to deal with PeptideSequence column 
 o filter_on_max_peptides.R: add ability to deal with PeptideSequence column


CHANGES IN VERSION 1.17.1
-------------------------
UPDATE

 o convert_protein_ids: count unique non-mapped IDs.

CHANGES IN VERSION 1.17.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.11 development release
 

CHANGES IN VERSION 1.16.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.10 release


CHANGES IN VERSION 1.15.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.10 development release
 

CHANGES IN VERSION 1.14.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.9 release


CHANGES IN VERSION 1.13.6
-------------------------
UPDATE

 o aLFQ is deprecated on CRAN and thus Vignette cannot depend anymore on the aLFQ package.
 

CHANGES IN VERSION 1.13.5
-------------------------
NEW FEATURES

 o Update plot_variation and plot_variation_vs_total function
 
 
CHANGES IN VERSION 1.13.4
-------------------------
NEW FEATURES

 o Update version for JPP (new version of pyprophet)
 

CHANGES IN VERSION 1.13.3
-------------------------
UPDATE

 o DESCRIPTION: move PECA to enhances 


CHANGES IN VERSION 1.13.2
-------------------------
UPDATE

 o Vignette: make PECA chunk eval=FALSE to prevent error when the PECA package is not available on BioC 


CHANGES IN VERSION 1.13.1
-------------------------
NEW FEATURES

 o Add ImmunoOncology to biocViews
 

CHANGES IN VERSION 1.13.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.9 development release
 

CHANGES IN VERSION 1.12.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.8 release

CHANGES IN VERSION 1.11.8
-------------------------
UPDATE

 o updating information on checking transitions in convert4aLFQ function.


CHANGES IN VERSION 1.11.7
-------------------------
UPDATE

 o avoid sorting when adding gene symbols in add_gene_symbol


CHANGES IN VERSION 1.11.6
-------------------------
BUG FIXES

 o fix manual page and function in convert_protein_ids to copy non-converted IDs


CHANGES IN VERSION 1.11.5
-------------------------
UDPATE

 o move from bioclite to BiocManager


CHANGES IN VERSION 1.11.4
-------------------------
BUG FIXES

 o fix manual page to convert_protein_ids


CHANGES IN VERSION 1.11.3
-------------------------
BUG FIXES

 o updates to convert_protein_ids, see that convert4aLFQ outputs character vectors


CHANGES IN VERSION 1.11.2
-------------------------
NEW FEATURES

 o add functions convert_protein_ids, load_mart, add_genesymbol

CHANGES IN VERSION 1.11.1
-------------------------
BUG FIXES

 o remove links from manual page SWATH2stats-package


CHANGES IN VERSION 1.11.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.8 development release
 

CHANGES IN VERSION 1.10.0
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.7 release

CHANGES IN VERSION 1.9.3
-------------------------
BUG FIXES

 o removed aLFQ direct link in manual


CHANGES IN VERSION 1.9.2
-------------------------
NEW FEATURES

 o add option check_tranisitions to convert_aLFQ function
 o replace align_orig_filename column with filename


CHANGES IN VERSION 1.9.1
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.7 development release
 

CHANGES IN VERSION 1.8.1
-------------------------
NEW FEATURES

 o SWATH2stats in BioC 3.6 release
 

CHANGES IN VERSION 1.7.7
-------------------------
BUG FIXES

 o update vignette to prevent misunderstanding what data to load


CHANGES IN VERSION 1.7.6
-------------------------
BUG FIXES

 o manual convert4PECA

CHANGES IN VERSION 1.7.5
-------------------------
BUG FIXES

 o library not recognized in vignette

CHANGES IN VERSION 1.7.4
-------------------------
NEW FEATURES

 o add function convert4PECA

CHANGES IN VERSION 1.7.3
-------------------------
NEW FEATURES

 o add option to keep all columns to disaggregate function


CHANGES IN VERSION 1.7.2
-------------------------
NEW FEATURES

 o add option to no not remove decoys to filter_proteotypic_peptides(), 
   filter_on_max_peptides(), and filter_on_min_peptides()

CHANGES IN VERSION 1.7.1
-------------------------
NEW FEATURES


R/convert_aLFQ.R
#' Convert table into the format for aLFQ
#'
#' This functions selects the columns necessary for the aLFQ R package.
#'
#' @param data A data frame containing the SWATH data in transition-level format
#' @param annotation Option to indicate if the data has been annotated, i.e. if
#'   the columns Condition, Replicate, Run are present. If option is set to true
#'   it will write a new run_id as a string of the combination of these three
#'   columns.
#' @param  check_transitions if number of transitions should be checked. As input only
#'   transition-level data should be used and therefore this is
#'   checked. However, this makes the function slow and herewith be omitted.
#' @return Returns a data frame in the appropriate format for aLFQ.
#' @references Rosenberger G, Ludwig C, Rost HL, Aebersold R, Malmstrom L. aLFQ:
#'   an R-package for estimating absolute protein quantities from label-free
#'   LC-MS/MS proteomics data. Bioinformatics. 2014 Sep 1;30(17):2511-3. doi:
#'   10.1093/bioinformatics/btu200.
#' @author Peter Blattmann
#' @examples
#'  data("OpenSWATH_data", package="SWATH2stats")
#'  data("Study_design", package="SWATH2stats")
#'  data <- sample_annotation(OpenSWATH_data, Study_design)
#'  data.filtered.decoy <- filter_mscore(data, 0.01)
#'  raw <- disaggregate(data.filtered.decoy)
#'  data.aLFQ <- convert_aLFQ(raw)
#' @export
convert_aLFQ <- function(data, annotation=TRUE, check_transitions=TRUE) {

  if (isTRUE(annotation)) {
    data <- data[, c("proteinname", "peptidesequence", "fragmention",
                     "nakedsequence", "precursorcharge", "intensity",
                     "condition", "bioreplicate", "run")]
    data[["run_id"]] <- paste(data[["condition"]], data[["bioreplicate"]], data[["run"]], sep="_")
  }

  colnames(data) <- gsub("proteinname", "protein_id", colnames(data))
  colnames(data) <- gsub("precursorcharge", "precursor_charge", colnames(data))
  colnames(data) <- gsub("fragmention", "transition_id", colnames(data))
  colnames(data) <- gsub("intensity", "transition_intensity", colnames(data))
  colnames(data) <- gsub("peptidesequence", "peptide_id", colnames(data))
  colnames(data) <- gsub("nakedsequence", "peptide_sequence", colnames(data))

  data[["transition_id"]] <- gsub("_run[[:digit:]]*$", "", data[["transition_id"]])
  data[["transition_id"]] <- paste(data[["peptide_sequence"]], data[["transition_id"]], sep=" ")

  data[["concentration"]] <- "?"

  data <- data[, c("run_id", "protein_id", "peptide_id", "transition_id", "peptide_sequence",
                   "precursor_charge", "transition_intensity", "concentration")]
  #check transitions
  data[["protein_id"]] <- factor(data[["protein_id"]])
  data[["peptide_id"]] <- factor(data[["peptide_id"]])
  data[["transition_id"]] <- factor(data[["transition_id"]])

  if (check_transitions) {
    data.agg <- aggregate(data[, c("transition_id")],
                          by=list(data[["peptide_id"]], data[["run_id"]]), length)
    if (median(data.agg[["x"]]) == 1) {
      warning("The aLFQ package should only be used with transition-level data.
              The data only contains one transition per peptide.")
    }
  }

  # convert back to character vector
  data[["protein_id"]] <- as.character(data[["protein_id"]])
  data[["peptide_id"]] <- as.character(data[["peptide_id"]])
  data[["transition_id"]] <- as.character(data[["transition_id"]])
  return(data)
}
